py/genv.py

```diff
@@ -112,36 +127,56 @@ def access_json(filename: str, factory: Callable[[], Dict], *, convert: Optional
             with open(path, 'w', opener=lambda path, flags: os.open(path, flags, 0o666)) as f:
                 json.dump(o, f, indent=4)
 
+
+# Convertors
+
 def memory_to_bytes(cap: str) -> int:
     """
     Convert memory string to an integer value in bytes.
     """
-    for unit, multiplier in [
-        ('b', 1),
-        ('k', 1000),
-        ('m', 1000 * 1000),
-        ('g', 1000 * 1000 * 1000),
-        ('ki', 1024),
-        ('mi', 1024 * 1024),
-        ('gi', 1024 * 1024 * 1024),
-    ]:
+    for unit, multiplier in MEMORY_TO_BYTES_MULTIPLIERS_DICT.items():
         if cap.endswith(unit):
             return int(cap.replace(unit, '')) * multiplier
 
     return int(cap)  # the value is already in bytes if no unit was specified
 
+
 def bytes_to_memory(bytes: int, unit: str) -> str:
     """
     Convert bytes to a memory string.
     """
-    multipliers = {
-        'b': 1,
-        'k': 1000,
-        'm': 1000 * 1000,
-        'g': 1000 * 1000 * 1000,
-        'ki': 1024,
-        'mi': 1024 * 1024,
-        'gi': 1024 * 1024 * 1024,
-    }
-
-    return f'{bytes // multipliers[unit]}{unit}'
+    return f'{bytes // MEMORY_TO_BYTES_MULTIPLIERS_DICT[unit]}{unit}'
+
+
+# Time functions
+
+
+def time_since(dt: Union[str, datetime]) -> str:
+    """
+    This function returns a human readable string describing the amount of time passed since 'dt'
+    :param dt: The base time to calculate from. Can be either string or datetime
+    :return: a human readable string describing the amount of time passed since 'dt'
+    """
+    if isinstance(dt, str):
+        dt = datetime.strptime(dt, DATETIME_FMT)
+
+    value = int((datetime.now() - dt).total_seconds())
+    unit = 'second'
+
+    for amount, next_units in [
+        (0, 'second'),
```

Comment:
out of curiousity, is this `(0, 'second')` de-facto used, or is it more for the sake of completeness?

Reply:
This was part of a small test I did when I though about the function refactor. Removed.